runtime/src/accounts_db.rs

```diff
@@ -1662,6 +1910,87 @@ pub mod tests {
         assert!(check_storage(&daccounts, 1, 10));
     }
 
+    fn assert_load_account(
+        accounts: &AccountsDB,
+        slot: Slot,
+        pubkey: Pubkey,
+        expected_lamports: u64,
+    ) {
+        let ancestors = vec![(slot, 0)].into_iter().collect();
+        let owner = Account::default().owner;
+        let (account, slot) = accounts
+            .load_slow(&ancestors, &pubkey)
+            .unwrap_or((Account::new(0, 0, &owner), slot));
+        assert_eq!((account.lamports, slot), (expected_lamports, slot));
+    }
+
+    fn reconstruct_accounts_db_via_serialization(accounts: AccountsDB, slot: Slot) -> AccountsDB {
+        let mut writer = Cursor::new(vec![]);
+        serialize_into(&mut writer, &AccountsDBSerialize::new(&accounts, slot)).unwrap();
+
+        let buf = writer.into_inner();
+        let mut reader = BufReader::new(&buf[..]);
+        let daccounts = AccountsDB::new(None);
+
+        let local_paths = {
+            let paths = daccounts.paths.read().unwrap();
+            AccountsDB::format_paths(paths.to_vec())
+        };
+
+        let copied_accounts = TempDir::new().unwrap();
+        // Simulate obtaining a copy of the AppendVecs from a tarball
+        copy_append_vecs(&accounts, copied_accounts.path()).unwrap();
+        daccounts
+            .accounts_from_stream(&mut reader, local_paths, copied_accounts.path())
+            .unwrap();
+
+        daccounts
+    }
+
+    fn purge_zero_lamport_accounts(accounts: &AccountsDB, slot: Slot) {
+        let ancestors = vec![(slot as Slot, 0)].into_iter().collect();
+        accounts.purge_zero_lamport_accounts(&ancestors);
+    }
+
+    #[test]
+    fn test_accounts_db_serialize_zero_and_free() {
```

Comment:
[It passes.](https://buildkite.com/solana-labs/solana/builds/15689#487679ee-0afc-4d15-bd2f-dceb8a9b3a9f/79-2464) :)

Reply:
[Even after rebase to the current master HEAD](https://buildkite.com/solana-labs/solana/builds/15691#1bfb21c6-7230-4eb4-b1eb-4d90eba3da4a/82-2899). :)